avocado/utils/build.py

```diff
@@ -33,17 +33,17 @@ def run_make(path, make='make', extra_args='', process_kwargs=None):
     os.chdir(path)
     cmd = make
 
-    env = {}
+    if process_kwargs is None:
+        process_kwargs = {}
     # Set default number of jobs as ncpus + 1
-    if "-j" not in os.environ.get("MAKEFLAGS", ""):
+    os_makeflags = os.environ.get('MAKEFLAGS', '')
+    if process_kwargs.get('env') is None:
+        process_kwargs['env'] = {}
+    args_makeflags = process_kwargs['env'].get('MAKEFLAGS', '')
+
+    if '-j' not in os_makeflags and '-j' not in args_makeflags:
         jobs = multiprocessing.cpu_count() + 1
-        env = process_kwargs.get('env', {})
-        if not env:
-            env = {"MAKEFLAGS": "-j%s" % jobs}
-        elif "-j" not in env:
-            env["MAKEFLAGS"] = "-j%s" % jobs
-    if env and process_kwargs['env'] is not None:
-        process_kwargs['env'].update(env)
+        process_kwargs['env'].update({'MAKEFLAGS': '-j%s' % jobs})
```

Comment:
This would override `MAKEFLAGS` in case they exist, that's why in my code was the `+=`...

I mean `run_make(path, process_kwargs={"env": {"MAKEFLAGS": "foo"}})` would ignore the `"foo"` MAKEFLAG.

Reply:
indeed. so all we need is to always concatenate:
```python
    if process_kwargs is None:
        process_kwargs = {}
        
    os_makeflags = os.environ.get('MAKEFLAGS', '')
    if process_kwargs.get('env') is None:
        process_kwargs['env'] = {}
    args_makeflags = process_kwargs['env'].get('MAKEFLAGS', '')

    if '-j' not in os_makeflags and '-j' not in args_makeflags:
        jobs = multiprocessing.cpu_count() + 1
        args_makeflags += ' -j%s' % jobs
        process_kwargs['env'].update({'MAKEFLAGS': args_makeflags})
```

Reply:
makes sense